### PR TITLE
Support tls

### PR DESCRIPTION
**Pull request by johnvenn**

### CODE REVIEW

File: Linux/sgx/libsgx_tsgxssl/tsgxssl_version.cpp [hunk 1]
```diff
@@ -30,7 +30,7 @@
  */
 
 // This string should be updated before every release!!
-#define STRFILEVER "1.1.1o"
+#define STRFILEVER "1.1.1s"
```
Comment (ShundaZhang) [inline Linux/sgx/libsgx_tsgxssl/tsgxssl_version.cpp:33]: Should be 1.1.1t? Same comment for usgxssl_version.cpp.

File: Linux/sgx/Makefile [hunk 2]
```diff
@@ -72,7 +72,6 @@ endif
 endif
 
 $(PACKAGE_LIB)/$(OPENSSL_LIB):
-	chmod 755 $(BUILD_OPENSSL)
```
Comment (ShundaZhang) [inline Linux/sgx/Makefile:75]: Why remove this chmod command, is it intentional?

File: Linux/sgx/libsgx_tsgxssl/tsgxssl_version.cpp (hunk 1 above)
Comment (jinghe-INTC) [reply Linux/sgx/libsgx_tsgxssl/tsgxssl_version.cpp:33]: Yes, it should be. I missed this on the master branch. 
@johnvenn Please correct it. And I will fix it on the master branch.

File: Linux/sgx/Makefile (hunk 2 above)
Comment (jinghe-INTC) [reply Linux/sgx/Makefile:75]: It seems unnecessary, if the code was checked out directly checked out , or decompressed from the code tar.gz package in Linux environment.

Comment (johnvenn) [reply Linux/sgx/Makefile:75]: > Why remove this chmod command, is it intentional?

the code was auto-merged from the master branch. Let me check the details. It is under testing.

File: Linux/sgx/libsgx_tsgxssl/tsgxssl_version.cpp (hunk 1 above)
Comment (johnvenn) [reply Linux/sgx/libsgx_tsgxssl/tsgxssl_version.cpp]: > Yes, it should be. I missed this on the master branch. @johnvenn Please correct it. And I will fix it on the master branch.

fixed in the PR.

File: Linux/sgx/Makefile (hunk 2 above)
Comment (johnvenn) [reply Linux/sgx/Makefile:75]: > Why remove this chmod command, is it intentional?

confirmed with Jing. That command was intentionally removed in the master branch sometime ago.

Comment (johnvenn) [reply Linux/sgx/Makefile:75]: test passed. @jinghe-INTC @ShundaZhang

**Pull request merged**